src/main/groovy/3mainConfiguration.groovy

```diff
@@ -0,0 +1,112 @@
+/**
+* main configuration
+*
+* This script will automate the configuration of the main jenkins options,
+* found at <jenkinsURL>/configure. It does not configure any of the options
+* offered via plugins.
+*
+* Values for this configuration should NEVER be placed in this script, but
+* available to Jenkins as an environment variable pointing to YAML data
+*
+**/
+
+import java.util.logging.Logger
+import jenkins.*
+import jenkins.model.*
+import hudson.model.*
+import hudson.tasks.Shell
+import hudson.slaves.EnvironmentVariablesNodeProperty
+import hudson.slaves.EnvironmentVariablesNodeProperty.Entry
+import hudson.model.Node.Mode
+@Grapes([
+    @Grab(group='org.yaml', module='snakeyaml', version='1.17')
+])
+import org.yaml.snakeyaml.Yaml
+
+Logger logger = Logger.getLogger("")
+Jenkins jenkins = Jenkins.getInstance()
+Yaml yaml = new Yaml()
+
+String configPath = System.getenv("JENKINS_CONFIG_PATH")
+String configText = ''
+try {
+    configText = new File("${configPath}/main_config.yml").text
+} catch (FileNotFoundException e) {
+    logger.severe("Cannot find config file path @ ${configPath}/main_config.yml")
+    jenkins.doSafeExit(null)
+    System.exit(1)
+}
+
+Map mainConfig = yaml.load(configText).MAIN
+Map propertiesConfig = yaml.load(configText).GLOBAL_PROPERTIES
+Map locationConfig = yaml.load(configText).LOCATION
+Map shellConfig = yaml.load(configText).SHELL
+
+logger.info("Configuring basic Jenkins options")
+try {
+    jenkins.setSystemMessage(mainConfig.SYSTEM_MESSAGE)
+    jenkins.setNumExecutors(mainConfig.NUMBER_OF_EXECUTORS)
+    jenkins.setLabelString(mainConfig.LABELS.join(' '))
+    jenkins.setQuietPeriod(mainConfig.QUIET_PERIOD)
+    jenkins.setScmCheckoutRetryCount(mainConfig.SCM_RETRY_COUNT)
+    jenkins.setDisableRememberMe(mainConfig.DISABLE_REMEMBER_ME)
+    if (mainConfig.USAGE == 'NORMAL') {
+        jenkins.setMode(Mode.NORMAL)
+    } else if (mainConfig.USAGE == 'EXCLUSIVE') {
+        jenkins.setMode(Mode.EXCLUSIVE)
+    }
+    else {
+        logger.severe('Invalid value specified for X')
```

Comment:
Curious about the string here. Is this going to come after another log entry so the person troubleshooting knows what "X" means?